keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NetflixSemVerVersioningStrategy.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.artifacts
+
+import com.netflix.spinnaker.keel.api.artifacts.PublishedArtifact
+import com.netflix.spinnaker.keel.api.artifacts.VersioningStrategy
+import com.netflix.spinnaker.keel.core.NETFLIX_SEMVER_COMPARATOR
+
+/**
+ * A [VersioningStrategy] that codifies the versioning scheme conventions at Netflix.
+ */
+object NetflixSemVerVersioningStrategy : VersioningStrategy {
+  override val type: String = "netflix-semver"
+
+  override val comparator: Comparator<String> =
+    NETFLIX_SEMVER_COMPARATOR
+
+  private val NPM_VERSION_REGEX = Regex("(\\d+\\.\\d+\\.\\d+(-rc(\\.\\d+)?)?)(-[h]?\\w+)?(-\\w+)?")
+  private val BUILD_REGEX = Regex("-[h]?")
+
+  /**
+   * Extracts a version display name from the longer version string, leaving out build and git details if present.
+   */
+  fun getVersionDisplayName(artifact: PublishedArtifact): String {
+    return NPM_VERSION_REGEX.find(artifact.version)?.groups?.get(1)?.value
+      ?: artifact.version
+  }
+
+  /**
+   * Extracts the build number from the version string, if available.
+   */
+  fun getBuildNumber(artifact: PublishedArtifact): Int? {
```

Comment:
aren't we get the build number straight from the event? like in the example you mentioned yesterday: ` build:
      id: 10` ?

Reply:
No, we're extracting it from the artifact version string.

Reply:
Confirmed: echo is not adding those fields to the events. We can add them later, though.